scripts/autotools/CheckCompiler.m4

```diff
@@ -11,6 +11,10 @@ fi
 AC_PROG_CC
 AC_PROG_CXX
 
+dnl AC_PROG_CC_C99 is previously called in config.m4, but AC_PROG_CC resets CC
+dnl so call it once more to ensure C99 remains enabled
+AC_PROG_CC_C99
```

Comment:
This file dates back to libmongoc's old Autotools configs. I'm not sure that it's still necessary, since we do inherit some context from PHP's own Autotools configuration, but this didn't seem like the time to remove the file entirely.